tests/unittests/utils/tabular/test_compression_utils.py

```diff
@@ -0,0 +1,70 @@
+import gzip
+import bz2
+import lzma
+
+from autogluon.utils.tabular.utils import compression_utils
+
+
+def test_get_validated_path_no_compression_fn():
+    no_extension_filepath = 'dummy_file'
+    assert(compression_utils.get_validated_path(no_extension_filepath) == no_extension_filepath)
+
+    single_extension_filepath = 'dummy_file.pkl'
+    assert(compression_utils.get_validated_path(single_extension_filepath) == single_extension_filepath)
+
+    multiple_extension_filepath = 'dummy_file.fake-foo.zip.pkl'
+    assert(compression_utils.get_validated_path(multiple_extension_filepath) == multiple_extension_filepath)
+
+
+def test_get_validated_path_with_compression_fn():
+    compression_fns = ['gzip', 'lzma', 'bz2']
+
+    expected_gzip_filepath = "dummy_file.gz"
+    expected_lzma_filepath = "dummy_file.lzma"
+    expected_bz2_filepath = "dummy_file.bz2"
+
+    no_extension_filepath = 'dummy_file'
+    assert(compression_utils.get_validated_path(no_extension_filepath, compression_fns[0])
+           == expected_gzip_filepath)
+    assert(compression_utils.get_validated_path(no_extension_filepath, compression_fns[1])
+           == expected_lzma_filepath)
+    assert(compression_utils.get_validated_path(no_extension_filepath, compression_fns[2])
+           == expected_bz2_filepath)
+
+    single_extension_filepath = 'dummy_file.pkl'
+    assert(compression_utils.get_validated_path(single_extension_filepath, compression_fns[0])
+           == expected_gzip_filepath)
+    assert(compression_utils.get_validated_path(single_extension_filepath, compression_fns[1])
+           == expected_lzma_filepath)
+    assert(compression_utils.get_validated_path(single_extension_filepath, compression_fns[2])
+           == expected_bz2_filepath)
+
+    multiple_extension_filepath = 'dummy_file.fake-foo.zip.pkl'
```

Comment:
happy to discuss, but is this what we want? If someone specifies they want to save a file named 'abc.def.pkl', this will currently save it as 'abc.pkl' with no compression. This seems a bit counter-intuitive. Is the idea to avoid having multiple extensions? If so, why? Would it make more sense to throw an exception if sent multiple extensions/any extensions as input?

Reply:
No, I misunderstood your previous comment about testing the `abc.def.pkl` filename, I think it does make sense to just append the compression function extensions and keep multiple extensions if they are used. So, `abc.def.pkl.gz` for a gzip compressed file named `abc.def.pkl`.  The extension shouldn't be changed if there is no compression though, the original filename should be saved.

Reply:
Ok makes sense! Regarding your last statement, I agree that original name should be saved, so instead of changing the dict for gz/bzip/etc., just change None to be '' instead of '.pkl'. I believe that is what you are saying?

Reply:
Yes, the way I had it now was in `get_validated_path` if compression_fn is None, it just returns the path unchanged: https://github.com/adrienatallah/autogluon/blob/3447591a2c6f8517cb3b4493181f034efb4abde9/autogluon/utils/tabular/utils/compression_utils.py#L30, so really the '.pkl' for None in the dict wasn't being used, which isn't the best.  I'll change the extension for None to be ' ' instead of '.pkl' and just always append the extensions.